File: Selection sorting algorithm/SelectionSort.cpp
#include <iostream>
using namespace std;

int main()
{
    int arr[] = {22, 15, 10, 7, 3, 1};
    int n = sizeof(arr) / sizeof(arr[0]);

    for (int i = 0; i < n; i++)
    {
        int min_idx = i;
        for (int j = i + 1; j < n; j++)
        {
            if (arr[min_idx] > arr[j])
            {
                min_idx = j;
            }
        }
        int minimum = arr[min_idx];
        int key = arr[i];
        arr[min_idx] = key;
        arr[i] = minimum;
    }

    for (int k = 0; k < n; k++)
    {
        cout << arr[k] << " ";
    }
    cout << endl;
}

File: Selection sorting algorithm/SelectionSort.java
public class SelectionSort {

    public static int[] sort(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n; i++) {
            int min_idx = i;
            for (int j = i + 1; j < n; j++) {
                if (arr[min_idx] > arr[j]) {
                    min_idx = j;
                }
            }
            int key = arr[i];
            int minimum = arr[min_idx];
            arr[i] = minimum;
            arr[min_idx] = key;
        }
        return arr;
    }

    public static void printArr(int[] arr) {
        int n = arr.length;
        for (int i = 0; i < n; i++) {
            System.out.print(arr[i] + " ");
        }
        System.out.println();
    }

    public static void main(String[] args) {
        int[] sample_arr = { 64, 25, 12, 22, 11 };
        int[] sorted_arr = sort(sample_arr);
        printArr(sorted_arr);
    }
}
